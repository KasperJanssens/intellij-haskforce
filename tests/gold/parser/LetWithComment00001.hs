module LetWithComment00001 where

f = let a  {-Comment, comment?-}= 7
        b = 9
     in a +b