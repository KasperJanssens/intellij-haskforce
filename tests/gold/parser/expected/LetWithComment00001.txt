Haskell File
  MODULEDECL
    PsiElement(HaskellTokenType.module)('module')
    QCONID
      CONID
        PsiElement(HaskellTokenType.conidRegexp)('LetWithComment00001')
    PsiElement(HaskellTokenType.where)('where')
  BODY
    FUNORPATDECL
      PAT
        VARID
          PsiElement(HaskellTokenType.varidRegexp)('f')
      RHS
        PsiElement(HaskellTokenType.=)('=')
        EXP
          LETEXP
            PsiElement(HaskellTokenType.let)('let')
            FUNORPATDECL
              PAT
                VARID
                  PsiElement(HaskellTokenType.varidRegexp)('a')
              RHS
                PsiElement(HaskellTokenType.=)('=')
                EXP
                  PsiElement(HaskellTokenType.integertoken)('7')
            PsiComment(HaskellTokenType.{-)('{-')
            PsiComment(HaskellTokenType.commenttext)('Comment, comment?')
            PsiComment(HaskellTokenType.-})('-}')
            FUNORPATDECL
              PAT
                VARID
                  PsiElement(HaskellTokenType.varidRegexp)('b')
              RHS
                PsiElement(HaskellTokenType.=)('=')
                EXP
                  PsiElement(HaskellTokenType.integertoken)('9')
            PsiElement(HaskellTokenType.in)('in')
            EXP
              QVAR
                QVARID
                  VARID
                    PsiElement(HaskellTokenType.varidRegexp)('a')
              QOP
                QVAROP
                  QVARSYM
                    VARSYM
                      PsiElement(HaskellTokenType.+)('+')
              QVAR
                QVARID
                  VARID
                    PsiElement(HaskellTokenType.varidRegexp)('b')